File: app/src/test/java/com/msaggik/algorithmsdatastructures/java/ai/MultiLayerPerceptronTest.java
package com.msaggik.algorithmsdatastructures.java.ai;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

public class MultiLayerPerceptronTest {

    @Test
    public void testPredict() {
        MultiLayerPerceptron mlp = new MultiLayerPerceptron(2, 3, 1, 0.1);
        double[] input = {0.5, 0.2};
        double[] output = mlp.predict(input);

        // Проверяем, что выход имеет правильный размер
        assertEquals(1, output.length);

        // Проверяем, что выход находится в диапазоне [0, 1]
        assertTrue(output[0] >= 0 && output[0] <= 1);
    }

    @Test
    public void testWeightsInitialization() {
        MultiLayerPerceptron mlp = new MultiLayerPerceptron(2, 3, 1, 0.1);

        // Проверяем, что веса не равны нулю
        for (double[] weights : mlp.weightsInputHidden) {
            for (double weight : weights) {
                assertNotEquals(0, weight, 0.01);
            }
        }

        for (double[] weights : mlp.weightsHiddenOutput) {
            for (double weight : weights) {
                assertNotEquals(0, weight, 0.01);
            }
        }
    }
}

File: app/src/test/java/com/msaggik/algorithmsdatastructures/kotlin/ai/MultiLayerPerceptronTest.kt
package com.msaggik.algorithmsdatastructures.kotlin.ai

import org.junit.Assert.*
import org.junit.Test

class MultiLayerPerceptronTest {

    @Test
    fun testPredict() {
        val mlp = MultiLayerPerceptron(2, 3, 1, 0.1)
        val input = doubleArrayOf(0.5, 0.2)
        val output = mlp.predict(input)

        // Проверяем, что выход имеет правильный размер
        assertEquals(1, output.size)

        // Проверяем, что выход находится в диапазоне [0, 1]
        assertTrue(output[0] in 0.0..1.0)
    }

    @Test
    fun testWeightsInitialization() {
        val mlp = MultiLayerPerceptron(2, 3, 1, 0.1)

        // Проверяем, что веса не равны нулю
        for (weights in mlp.weightsInputHidden) {
            for (weight in weights) {
                assertNotEquals(0.0, weight, 0.01)
            }
        }

        for (weights in mlp.weightsHiddenOutput) {
            for (weight in weights) {
                assertNotEquals(0.0, weight, 0.01)
            }
        }
    }
}